### ISSUE TITLE
How to solve some problems encountered when reproducing experimental results?

### ISSUE THREAD

**Issue by aiwensiler** [RL4VLM/RL4VLM#16]
During my attempt to reproduce the experimental results of the paper, I encountered some issues and would appreciate your help. Some of the problems are as follows:

1. Firstly, there's a font issue. Several simulation environments in this project use "dejavu/DejaVuSans-Bold.ttf". Since the project files do not include this font file, I have downloaded a range of official DejaVuSans fonts such as "DejaVuSans-Bold.ttf" and "DejaVuSans.ttf". I also tried using the "Minecraft.ttf" font that comes with the project. Although these can be used, the displayed fonts still differ significantly from the fonts on the images in the training set provided for SFT. This discrepancy may affect the model's ability to distinguish different fonts after SFT, causing it to fail in correctly recognizing the digits in the images. Could you please provide the exact font file used during the generation of the training set?

2. In the numberline and ezpoints environments, during the SFT (unadjusted parameters) phase, the loss rapidly decreases in the first few training sessions and then fluctuates slightly. The reason might be due to a high learning rate or the dataset being relatively small and highly repetitive. While it might be more optimal to complete a full training cycle for large model training, it is currently unclear if this affects the subsequent model performance. However, the aforementioned issue of low font distinction ability after SFT does occur. Is this situation normal?

3. In the numberline and ezpoints environments (other environments are still being tested), both the llava-1.6-mistral-7b model without SFT and the model with SFT (unadjusted parameters) experience a sharp decline in accuracy shortly after a brief increase during the RL step (unadjusted parameters). The responses from the model also become increasingly erratic, sometimes resulting in no response at all. (In particular, the model performance after SFT is worse than that of the model without SFT, possibly due to the font issue mentioned above). What could be causing this issue? I will try using the SFT model you suggested in other issues to see if it helps.

**Comment by aiwensiler** [comment]
Okay, for the first issue, I think I have found the corresponding font. However, I still recommend that you add the font file to the project.

**Comment by YX-S-Z** [comment]
Hi, thank you very much for your questions.

Regarding the font size issues, we apologize for the confusion, and we have provided a version of the font from the server we were using.  You may also want to download them from [here](https://www.fontsquirrel.com/fonts/dejavu-sans) if you find it helpful.

Regarding SFT, you may directly start from the checkpoint we posted in an earlier issue, and see if that is helpful, with the updated font. 

Regarding your observations: 

> Without SFT and the model with SFT (unadjusted parameters) experience a sharp decline in accuracy shortly after a brief increase during the RL step (unadjusted parameters)

We do observe this phenomenon a lot during our development, especially if we directly start from the `llava-1.6-mistral` checkpoint, and this issue gets largely resolved from the sft checkpoint. Our hypothesis for this phenomenon is that, `llava-1.6-mistral` has not been trained on `json` instruction following format, hence directly start from `llava-1.6-mistral` with RL training will make the model hard to follow the specific `json` format, but we have nor thoroughly studied this hypothesis yet. 

You may also want to try [Cambrian-1](https://github.com/cambrian-mllm/cambrian) as the backbone VLM, which has been trained on `json` instruction following data. We have tested our framework on Cambrian-1 and it works (but we may need additional time to update the codebase to incorporate Cambrian-1). 

Please let us know if you have more questions!
